include/open62541/server.h

```diff
@@ -1285,6 +1285,114 @@ UA_Server_triggerEvent(UA_Server *server, const UA_NodeId eventNodeId, const UA_
+                          const UA_NodeId hierarchialReferenceType, UA_NodeId *outConditionId);
+
+/**
+ * set the value of condition field.
+ * @param server The server object
+ * @param condition The NodeId of the node representation of the Condition Instance
+ * @param variantValue Variant Value to be written to the Field
+ * @param type Variant Type.
+ * @param fieldName Name of the Field in which the value should be written
+ * @return The StatusCode of the UA_Server_triggerEvent method*/
+UA_StatusCode UA_EXPORT
+UA_Server_setConditionField(UA_Server *server, const UA_NodeId condition,
+                            const void* variantValue, UA_UInt16 type,
+                            const UA_QualifiedName fieldName);
+
+/**
+ * set the value of property of condition field.
+ * @param server The server object
+ * @param condition The NodeId of the node representation of the Condition Instance
+ * @param variantValue Variant Value to be written to the Field
+ * @param type Variant Type
+ * @param variableFieldName Name of the Field which has a property
+ * @param variablePropertyName Name of the Field Property in which the value should be written
+ * @return The StatusCode of the UA_Server_triggerEvent method*/
+UA_StatusCode UA_EXPORT
+UA_Server_setConditionVariableFieldProperty(UA_Server *server, const UA_NodeId condition,
+                                            const void *variantValue, UA_UInt16 type,
+                                            const UA_QualifiedName variableFieldName,
+                                            const UA_QualifiedName variablePropertyName);
+
+/**
+ * triggers an event only for an enabled condition. The condition list is updated then with the
+ * last generated EventId.
+ * @param server The server object
+ * @param condition The NodeId of the node representation of the Condition Instance
+ * @param conditionSource The NodeId of the node representation of the Condition Source
+ * @param outEventId last generated EventId
+ * @return The StatusCode of the UA_Server_triggerEvent method*/
+UA_StatusCode UA_EXPORT
+UA_Server_triggerConditionEvent(UA_Server *server, const UA_NodeId condition,
+                                const UA_NodeId conditionSource, UA_ByteString *outEventId);
+
+/**
+ * add an optional condition field using its name. (TODO Adding optional methods
+ * is not implemented yet)
+ * @param server The server object
+ * @param condition The NodeId of the node representation of the Condition Instance
+ * @param conditionType The NodeId of the node representation of the Condition Type
+ * from which the optional field comes
+ * @param fieldName Name of the optional field
+ * @param outOptionalVariable The NodeId of the created field (Variable Node)
+ * @return The StatusCode of the UA_Server_triggerEvent method*/
+UA_StatusCode UA_EXPORT
+UA_Server_addConditionOptionalField(UA_Server *server, const UA_NodeId condition,
```

Comment:
What is the difference between `UA_Server_writeObjectProperty_scalar` and `UA_Server_setConditionVariableFieldProperty`?

Reply:
`UA_Server_setConditionVariableFieldProperty ` is almost the same as `UA_Server_writeObjectProperty_scalar`. The difference is that the first sets a property of a Variable Node (in this case a Condition Field). It takes the Condition NodeId, Field Name and Field Property Name as input parameter. Using `UA_Server_writeObjectProperty_scalar` (for users) will require more code in my opinion, because all Fields are generated automatically when we create a Condition Node und we don't track the NodeIds of the created Fields. So the user should find the correct NodeIds first using their BrowseName then perform `UA_Server_writeObjectProperty_scalar`.

`UA_Server_setConditionField` and
`UA_Server_setConditionVariableFieldProperty ` use only public API functions, so the user can implement their own logic without problems. Inside the A&C module those function could be used staticly. What do you think is best?